File: New Year Resolution.cpp
class Solution {
    int help(int coins[], int n, int i, int sum) {
        if(sum != 0 && (sum % 20 == 0 || sum % 24 == 0 || sum == 2024)) return 1;
        if(i == n) return 0;
        
        int pick = help(coins, n, i+1, sum+coins[i]);
        if(pick == 1) return 1;
        int notPick = help(coins, n, i+1, sum);
        
        return pick || notPick;
    }
    
    public:
    int isPossible(int n, int coins[]) {
        return help(coins, n, 0, 0);
    }
    
};

File: New Year Resolution.java
class Solution {
    public static boolean isPossible(int N, int[] coins) {
        // code here
        int sum=0;
        for(int i:coins)sum+=i;
        boolean[][] dp=new boolean[N+1][sum+1];
        return f(0,0,coins,dp);
    }
    static boolean f(int i,int sum,int[] coins,boolean[][] dp){
        if(sum!=0 && (sum%20==0 || sum%24==0 || sum==2024))return true;
        if(i>=coins.length)return false;
        if(dp[i][sum])return dp[i][sum];
        return dp[i][sum] |=f(i+1,sum+coins[i],coins,dp)||f(i+1,sum,coins,dp);
    }
}
